### PR TITLE
Blog validations model spec

### PR DESCRIPTION
**Pull request by BonkeGcobo**
### Added validations for the user model  
 - [x] Validation name should not be blank and the test for it
 - [x] PostCounter should be integer and equal or greater than 0 and corresponding test
 
 ### Added validations for the Post Model
- [x] Post title should not be blank and the test for it
- [x] Title Characters should be a maximum of 250
- [x] Comments counter should be integer and zero or greater with corresponding test
- [x] Likes counter should be integer and zero or greater with corresponding test 

**Questions**
should these the only validations for should added for this branch, should I include for likes and comments as well?

### CODE REVIEW

**Comment by Munsa1** [comment]
HELLO, @BonkeGcobo  kindly remove section descriptions from your readme file so I can review your project. In case you do not know what they are, check the screenshot below

image


### You have 3 of them.

**Comment by Munsa1** [comment]
@BonkeGcobo Kindly remove the `Live Demo` section from your read me file and add `Fututre Features` to the Future Features section, coming soon is not enough, add something like `Add devise authentication` and `Add views`

File: README.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ For more information on the importance of a professional README for your reposit
 
 # 📖 [Blog_App] <a name="about-project"></a>
 
-> Describe your project in 1 or 2 sentences.
+
 
 **[BlogApp]** The Blog app will be a classic example of a blog website. It is a functional website that will show the list of posts and empower readers to interact with them by adding comments and liking posts. It has been made by the following ER diagram.
```
Comment (Munsa1) [inline README.md:70]: - [x] KIndly fix the broken link to the MIT licences agreement under the LIcense Section for a professional readme file

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  it 'Title should always be present' do
+    post = Post.new(author: @user, Title: 'Some Test', comments_counter: 0, likes_counter: 0)
+    expect(post).to be_valid
+  end
+
+  it 'Title should be a maximum of 250 chars' do
+    post = Post.new(author: @user, Title: 'A' * 251, comments_counter: 0, likes_counter: 0)
+    expect(post).to_not be_valid
```
Comment (Munsa1) [inline spec/models/post_spec.rb:15]: - [x] All custom methods in the User, Post, Comment, and Like models should be tested. Kindly do so. You have methods such as `recent_post` in the user model, `update_counter`, and `recent_comments` methods in the Post model to mention a few. Kindly ensure you add tests for all your models.

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  it 'comments counter should be integer' do
+    Post.new(author: @user, Title: 'A' * 251, comments_counter: 'string', likes_counter: 0)
+    expect(subject).to_not be_valid
+
+    Post.new(author: @user, Title: 'A' * 251, comments_counter: true, likes_counter: 0)
+    expect(subject).to_not be_valid
+
+    Post.new(author: @user, Title: 'A' * 251, comments_counter: 3.21, likes_counter: 0)
```
Comment (Munsa1) [inline spec/models/post_spec.rb:30]: ### Due to insufficient files changed I will leave a comment here.

- [ ] It is a requirement that we have unit tests for the `User`, `Post`, `Comment`, and `Like` models. You only have tests for two files, the Post and User model. Kindly add tests for the Like and Comment models as well.
image

File: spec/models/post_spec.rb (hunk 2 above)
Comment (Munsa1) [reply spec/models/post_spec.rb:15]: - [x] Also please do not forget to add tests for CUSTOM methods in the Like and Comment models as per the requirement.

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @BonkeGcobo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Passing Test Cases :heavy_check_mark: 
Added Validations :heavy_check_mark: 
Descriptive Pr :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Munsa1** [comment]
- [ ] Also consider removing the comment in the screenshot below for a more professional readme file
image

File: README.md [hunk 4]
```diff
@@ -265,7 +260,7 @@ I would like to thank...
 
 ## 📝 License <a name="license"></a>
 
-This project is [MIT](./LICENSE) licensed.
+This project is [MIT](./MIT) licensed.
 
 _NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
```
Comment (aamirkhan2478) [inline README.md]: - [x] As mentioned by the previous :point_up: please remove this line for more professional readme file but you didn't remove this line yet so kindly remove it :slightly_smiling_face:

**Review by aamirkhan2478 — CHANGES_REQUESTED** [review]
Hi @BonkeGcobo, 



Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

 ## Highlights
✔️ Good Effort

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @BonkeGcobo 

Good job implementing the suggested changes 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Implemented suggested changes
-  ✅ Great work using the right Gitflow 
-  ✅ Your README looks good
-  ✅ Linters are working properly

## Optional suggestions ❕
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**